### PR TITLE
Issue/41: Add win38 support 

### PR DESCRIPTION
**Pull request by jonzarecki**
Continuation of the great work in #85 (couldn't continue his branch, unfortunately).
I've merged the current version and applied additional fixes to make the tests run.
Changed version to 2.9.0-alpha.3 

This seems to be working for my team's usage for the last 2 weeks. 

Might be a good idea to squash my commits.

### CODE REVIEW

**Comment by jonzarecki** [comment]
@samj1912 I'm free to change the code if needed. Able to run the ci?

**Comment by jonzarecki** [comment]
Weird, alright I'll check it :)

**Comment by jonzarecki** [comment]
@samj1912 Done.
Notice that @juhuebner added pytest-repeat and repeats all tests 5 times in the windows gh-actions.
It's because as he stated previously windows have permission errors that don't always appear. It does take 5 minutes for the CI to run the tests, but I agree as well that windows problems might not appear in only one run.

**Comment by sharadpattanshetti** [comment]
Any ETA about releasing this feature?

File: scripts/test.sh [hunk 1]
```diff
@@ -1,5 +1,26 @@
 #!/bin/bash
 set -euxo pipefail
 
-poetry run pytest -s -n auto --cov=cruft/ --cov=tests --cov-report=term-missing ${@-} --cov-report xml --cov-report html
+# set nocasematch option
+shopt -s nocasematch
```
[the remark this answers is not in the document]
Comment (jonzarecki) [reply scripts/test.sh:5]: Kept `pragma: no cov_4_nix`, and added the --ci flag

File: cruft/_commands/check.py [hunk 2]
```diff
@@ -17,15 +17,15 @@ def check(
...
             cruft_state["template"],
             Path(cookiecutter_template_dir),
             checkout,
             filter="blob:none",
             no_checkout=True,
-        )
-        last_commit = repo.head.object.hexsha
+        ) as repo:
+            last_commit = repo.head.object.hexsha
         if utils.cruft.is_project_updated(repo, cruft_state["commit"], last_commit, strict):
```
[the remark this answers is not in the document]
Comment (samj1912) [reply cruft/_commands/check.py:29]: @jonzarecki?

[the remark this answers is not in the document]
Comment (jonzarecki) [reply cruft/_commands/check.py:29]: Forgot about that part, it works but it probably should.

**Comment by jonzarecki** [comment]
@samj1912 Done 👍🏻

**Review by samj1912 — APPROVED** [review]

**Comment by samj1912** [comment]
@jonzarecki Thank you so much for this PR and your contributions :)!

**Comment by jonzarecki** [comment]
@samj1912  Do we need to bump the version or re-publish to pypi?

**Pull request merged**